nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-jsf/src/main/java/org/nuxeo/ecm/admin/setup/SetupWizardActionBean.java

```diff
@@ -276,14 +267,19 @@ protected void saveParameters() {
 
         // Remove empty values for password keys
         for (String pwdKey : SECRET_KEYS) {
-            if (StringUtils.isEmpty(parameters.get(pwdKey))) {
+            if (StringUtils.isEmpty((String) parameters.get(pwdKey))) {
                 parameters.remove(pwdKey);
             }
         }
 
-        Map<String, String> customParameters = new HashMap<>();
-        customParameters.putAll(parameters);
-        customParameters.putAll(advancedParameters);
+        // compute <String, String> parameters for the ConfigurationGenerator
+        Stream<Entry<String, Serializable>> parametersStream = parameters.entrySet().stream();
+        Stream<Entry<String, Serializable>> advancedParametersStream = advancedParameters.entrySet().stream();
+        Map<String, String> customParameters = Stream.concat(parametersStream, advancedParametersStream).collect(
+                HashMap::new, (m, e) -> m.put(e.getKey(), Optional.ofNullable(e.getValue())
+                                                                  .map(Object::toString)
+                                                                  .orElse(null)),
+                HashMap::putAll);
```

Comment:
We tried this, and we can't use `Collectors.toMap` because `advancedParametersStream` owns `keys` already present in `parametersStream` which will thrown an exception if we use `Collectors.toMap`. Maybe add a comment ?